projects/super_res/super_res.py

```diff
@@ -0,0 +1,203 @@
+import xarray as xr
+import numpy as np
+import fv3viz
+from vcm.catalog import catalog
+import matplotlib.pyplot as plt
+import os
+import cartopy.crs as ccrs
+
+GRID = catalog["grid/c384"].read()
+
+DPI = 800
+
+
+def repeat_arr(arr, upscale_factor):
+    # This repeats values along the last dimension
+    return np.repeat(arr, upscale_factor, axis=-1)
+
+
+def upscale_nearest(ds, upscale_factor, dim):
+    ds_upscaled = xr.apply_ufunc(
+        repeat_arr,
+        ds,
+        input_core_dims=[[dim]],  # apply function along this dimension
```

Comment:
nit: I think that these comments are unnecessary. They are repeating the definition of core_dim.

Reply:
I don't expect Prakhar to know the definition of a `core_dim` as a new xarray user, so I added helpful comments like this. `apply_ufunc` in particular is pretty idiosyncratic. He can delete comments like this as desired, if he wants me to do that in this PR I can.

Reply:
I'd liek to leave the comments for now, they were super useful.